Sources/XcodeGenKit/PBXProjGenerator.swift

```diff
@@ -354,9 +354,11 @@ public class PBXProjGenerator {
         addObject(resourcesBuildPhase)
         buildPhases.append(resourcesBuildPhase.reference)
 
-        let headersBuildPhase = PBXHeadersBuildPhase(reference: generateUUID(PBXHeadersBuildPhase.self, target.name), files: getBuildFilesForPhase(.headers))
-        addObject(headersBuildPhase)
-        buildPhases.append(headersBuildPhase.reference)
+        if target.type == .framework || target.type == .dynamicLibrary {
+            let headersBuildPhase = PBXHeadersBuildPhase(reference: generateUUID(PBXHeadersBuildPhase.self, target.name), files: getBuildFilesForPhase(.headers))
```

Comment:
Should we also only create this if `getBuildFilesForPhase` here returns a  non-empty array?

Reply:
Yes, this could be added though the same would apply to the sources and resources build phase. I left them in there in case people wanted to easily files add to them in Xcode (that's assuming they are only generating once and then checking in the project file)
In either case it's not something we have to cover in this PR

Reply:
Ah makes sense, I guess for some of these it's just preference, since in our all swift project we'll _never_ need this build phase.

Reply:
Yeah, I think the empty ones could be removed

Reply:
This has been done in #149